tools/rc/usResourceCompiler.cpp

```diff
@@ -0,0 +1,523 @@
+  {0,0,0,0,0,0}
+};
+
+// ---------------------------------------------------------------------------------
+// -----------------------------    MAIN ENTRY POINT    ----------------------------
+// ---------------------------------------------------------------------------------
+
+int main(int argc, char** argv)
+{
+  int compressionLevel = 6;
+  int return_code = EXIT_SUCCESS;
+  std::string zipFile;
+  std::string bundleName;
+  std::string archiveName;
+  
+  argc -= (argc > 0);
+  argv += (argc > 0); // skip program name argv[0]
+  option::Stats stats(usage, argc, argv);
+  std::unique_ptr<option::Option[]> options(new option::Option[stats.options_max]);
+  std::unique_ptr<option::Option[]> buffer(new option::Option[stats.buffer_max]);
+  option::Parser parse(usage, argc, argv, options.get(), buffer.get());
+  
+  if (parse.error())
+  {
+    std::cerr << "Parsing command line arguments failed. " << std::endl;
+    return_code = EXIT_FAILURE;
+  }
+  
+  if (parse.nonOptionsCount())
+  {
+    std::clog << "unrecognized options ..." << std::endl;
+    for (int i = 0; i < parse.nonOptionsCount(); ++i)
+    {
+      std::cout << "\t" << parse.nonOption(i) << std::endl;
+    }
+    return_code = EXIT_FAILURE;
+  }
+  
+  option::Option* bundleFileOpt = options[BUNDLEFILE];
+  if (bundleFileOpt && bundleFileOpt->count() > 1 )
+  {
+    std::cerr << "(--append-binary | -a) appear multiple times in the arguments. Check usage." << std::endl;
+    return_code = EXIT_FAILURE;
+  }
+  
+  option::Option* outFileOpt = options[OUTFILE];
+  if (outFileOpt && outFileOpt->count() > 1 )
+  {
+    std::cerr << "(--out-file | -o) appear multiple times in the arguments. Check usage." << std::endl;
+    return_code = EXIT_FAILURE;
+  }
+  
+  if (!bundleFileOpt && !outFileOpt)
+  {
+    std::cerr << "At least one of the options (--bundle-file | --out-file) is required." << std::endl;
+    return_code = EXIT_FAILURE;
+  }
+  
+  if (argc == 0 || options[HELP] || return_code == EXIT_FAILURE)
+  {
+    option::printUsage(std::clog, usage);
+    return return_code;
+  }
+  
+  if (options[BUNDLENAME])
+  {
+    bundleName = options[BUNDLENAME].arg;
+  }
+  
+  if (!options[VERBOSE])
+  {
+    // if not in verbose mode, supress the clog stream
+    std::clog.setstate(std::ios_base::failbit);
+  }
+  
+  if (options[COMPRESSIONLEVEL])
+  {
+    char* endptr = 0;
+    compressionLevel = strtol(options[COMPRESSIONLEVEL].arg, &endptr, 10);
+  }
+  std::clog << "using compression level " << compressionLevel << std::endl;
+  
+  std::string outFile;
+  bool deleteTempFile = false;
+  if (outFileOpt)
+  {
+    outFile = outFileOpt->arg;
+  }
+  else
+  {
+    outFile = us_tempfile();
+    deleteTempFile = true;
+  }
+  
+  try
+  {
+    if (!options[RESADD] && !options[MANIFESTADD] && options[ZIPADD].count() == 1 && options[BUNDLEFILE])
```

Comment:
A short comment calling out the requirement that we only support appending one zip to a bundle would be helpful. 
The ```options[ZIPADD].count() == 1``` appears (from someone unfamiliar with the code and requirements) to be arbitrary. A "why" comment would be helpful.